### PR TITLE
Type tools/project_manager/remote_file_manager.php

### PR DESCRIPTION
**Pull request by jchaffraix**
TEST=Interacted with the remote file manager (uploaded/deleted files, check directories)

_Edit by srjfoo_: Sandbox at [~srjfoo/c.branch/julien_type_tools_remote_file_manager/](https://www.pgdp.org/~srjfoo/c.branch/julien_type_tools_remote_file_manager/)

### CODE REVIEW

File: tools/project_manager/remote_file_manager.php [hunk 1]
```diff
@@ -737,14 +737,14 @@ function is_valid_move_destination($dir)
  * removing empty components, rejecting parent traversal,
  * and re-joining them.
  *
- * Returns False is path contains invalid components.
+ * Returns null is path contains invalid components.
  */
-function canonicalize_path($relpath)
+function canonicalize_path(string $relpath): ?string
```
Comment (jchaffraix) [inline tools/project_manager/remote_file_manager.php:741]: This function and `get_directory_items_sorted` were switched from returning `false` on error to returning `null` as it yields to be a better type. I updated the callers, which were all in this file, to check against `null`.

File: tools/project_manager/remote_file_manager.php [hunk 2]
```diff
@@ -1075,33 +1078,31 @@ function show_message($type, $message)
 /**
  * Display a return link (to the 'showdir' view)
  */
-function show_return_link($relpath = null)
+function show_return_link(): void
```
Comment (jchaffraix) [inline tools/project_manager/remote_file_manager.php:1080]: None of the callers passed the only argument so I removed it and simplified the code below.

File: tools/project_manager/remote_file_manager.php [hunk 3]
```diff
@@ -204,14 +204,13 @@
     default:
         // no matching $action in input
         fatal_error(sprintf(_("Invalid action: '%s'"), html_safe($action)));
-        exit;
```
Comment (jchaffraix) [inline tools/project_manager/remote_file_manager.php:207]: This `exit` is unreachable as `fatal_error` doesn't return.

**Review by srjfoo — APPROVED** [review]

File: tools/project_manager/remote_file_manager.php [hunk 4]
```diff
@@ -737,14 +736,14 @@ function is_valid_move_destination($dir)
  * removing empty components, rejecting parent traversal,
  * and re-joining them.
  *
- * Returns False is path contains invalid components.
+ * Returns null is path contains invalid components.
```
[the remark this answers is not in the document]
Comment (jchaffraix) [reply tools/project_manager/remote_file_manager.php]: I don't know if I understood your question so I rephrased this sentence as it had a typo (`is` should have been `if`). Let me know if the updated phrasing clears up your confusion:

> Returns null if `$relpath` contains any invalid component.

**Review by chrismiceli — APPROVED** [review]

File: tools/project_manager/remote_file_manager.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (srjfoo) [reply tools/project_manager/remote_file_manager.php]: Yes, that clarifies it. It should have occurred to me that it might be an is/if typo. Thanks!

**Review by cpeel — APPROVED** [review]

**Review by 70ray — APPROVED** [review]

**Pull request merged**